providers/base/bin/camera_test.py

```diff
@@ -240,32 +265,95 @@ def _detect_and_show_camera_info(self, device, cp):
 
         return 0 if capture_capabilities else 1
 
-    def _stop(self):
-        self.camerabin.set_state(Gst.State.NULL)
-        Gtk.main_quit()
+    def _on_gst_message(self, bus, message):
+        # Process End-of-stream messages
+        if message.type == self.Gst.MessageType.EOS:
+            # Stop the pipeline
+            self.pipeline.set_state(self.Gst.State.NULL)
+            # Quit the GLib main loop
+            self.main_loop.quit()
+            if self.log_level == logging.DEBUG:
+                print("End-of-stream")
```

Comment:
Applies to most other places, we have configured `logging`, lets use it! 
```suggestion
            logging.debug("End-of-stream")
```

Reply:
Done!